### PR TITLE
[TACHYON-920] Added the publicApi annotation

### PR DESCRIPTION
**Pull request by yupeng9**
https://tachyon.atlassian.net/browse/TACHYON-920

So that developers are aware of the external facing APIs, and therefore ought be careful about changes on them.

In the future, we could build tools to monitor/audit the changes on all public APIs, generate reports of the changes, and then communicate them clearly with the users.

### CODE REVIEW

File: clients/unshaded/src/main/java/tachyon/client/RemoteBlockReader.java [hunk 1]
```diff
@@ -23,12 +23,14 @@
...
 
 import tachyon.Constants;
+import tachyon.annotation.PublicApi;
 import tachyon.conf.TachyonConf;
 import tachyon.util.CommonUtils;
 
 /**
  * The interface to read remote block from data server.
  */
+@PublicApi
```
Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/RemoteBlockReader.java]: why is this a public api?
are users supposed to use `RemoteBlockReader`?

File: clients/unshaded/src/main/java/tachyon/client/RemoteBlockWriter.java [hunk 2]
```diff
@@ -22,12 +22,14 @@
...
 
 import tachyon.Constants;
+import tachyon.annotation.PublicApi;
 import tachyon.conf.TachyonConf;
 import tachyon.util.CommonUtils;
 
 /**
  * The interface to write a remote block to the data server.
  */
+@PublicApi
```
Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/RemoteBlockWriter.java]: Also are users supposed to use `RemoteBlockWriter` directly?

**Comment by apc999** [comment]
Also I think these are public api:
tachyon.hadoop.TFS -- the hadoop compatible interface.
tachyon.client.file.FileInStream and
tachyon.client.file.FileOutStream

File: clients/unshaded/src/main/java/tachyon/client/RemoteBlockReader.java (hunk 1 above)
Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/client/RemoteBlockReader.java]: If I understand correctly, users can implement their own block reader/writer, right? @calvinjia

**Comment by yupeng9** [comment]
Thanks for the review. Will annotate those interfaces.

File: clients/unshaded/src/main/java/tachyon/client/RemoteBlockReader.java (hunk 1 above)
Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/RemoteBlockReader.java]: I think this is hidden from users who will only be using FileInStream/FileoutStream

File: clients/unshaded/src/main/java/tachyon/client/OutStream.java [hunk 3]
```diff
@@ -18,10 +18,13 @@
 import java.io.IOException;
 import java.io.OutputStream;
 
+import tachyon.annotation.PublicApi;
+
 /**
  * Provides a stream API to write to Tachyon. Only one OutStream should be opened for a Tachyon
  * object. This class is not thread safe and should only be used by one thread.
  */
+@PublicApi
```
Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/OutStream.java]: Revert this file?

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**